web/src/utils/metrics.ts

```diff
@@ -170,12 +170,18 @@ const parseTopologyMetric = (
   const sourceFields: Partial<TopologyMetricPeer> = {
     addr: raw.metric.SrcAddr,
     resource: nameAndType(raw.metric.SrcK8S_Name, raw.metric.SrcK8S_Type),
-    owner: nameAndType(raw.metric.SrcK8S_OwnerName, raw.metric.SrcK8S_OwnerType)
+    owner:
```

Comment:
why do we need to avoid setting owner if it's the same as leaf Type? Curious because it used to work like this even before the scopes refactor PR

Reply:
(I'm just worrying about potential side effects even if I don't see any right now...)

Reply:
There was tons of hardcoded logic for each case (when owner is set / when resource is set and so on), especially for grouping and displaying topology nodes.
Some of this logic is kept but while debugging, I saw that `Pods` and `Services` without owners where duplicating their name / type into the owner field.

That's kind of a nonsence and will avoid mistakes if you check the owner field first.

Reply:
The same logic was already hardcoded in the display btw:
https://github.com/netobserv/network-observability-console-plugin/blob/72915fe5ee0a9ddac8fde3ccbc02b824e920e23f/web/src/components/netflow-topology/element-fields.tsx#L35